server/manager/src/main/java/org/apache/accumulo/manager/tableOps/bulkVer2/LoadFiles.java

```diff
@@ -202,104 +212,127 @@ private Client(HostAndPort server, TabletClientService.Client service) {
       }
     }
 
-    private void sendQueued(int threshhold) {
-      if (queuedDataSize > threshhold || threshhold == 0) {
-        var sendTimer = Timer.startNew();
+    private void sendBackground() {
+      var queue = this.backgroundQueue.get();
+      List<Client> clients = new ArrayList<>();
 
-        List<Client> clients = new ArrayList<>();
-        try {
-
-          // Send load messages to tablet servers spinning up work, but do not wait on results.
-          loadQueue.forEach((server, tabletFiles) -> {
+      try {
+        var sendTimer = Timer.startNew();
 
-            if (log.isTraceEnabled()) {
-              log.trace("{} asking {} to bulk import {} files for {} tablets", fmtTid, server,
-                  tabletFiles.values().stream().mapToInt(Map::size).sum(), tabletFiles.size());
-            }
+        // Send load messages to tablet servers spinning up work, but do not wait on results.
+        queue.forEach((server, tabletFiles) -> {
```

Comment:
Does this need to be synchronized? I don't think so since it seems the queue under the atomic ref is only ever updated via `compareAndSet` but want to confirm

Reply:
The foreground thread abandons the map and never uses it again when it sets it on the atomic ref.  So when the background thread picks it up its the only thread using it.  So synchronization is not needed.